src/client/Client.cc

```diff
@@ -14071,7 +14071,9 @@ int Client::_removexattr(Inode *in, const char *name, const UserPerm& perms)
   if (strncmp(name, "user.", 5) &&
       strncmp(name, "security.", 9) &&
       strncmp(name, "trusted.", 8) &&
-      strncmp(name, "ceph.", 5))
+      strncmp(name, "ceph.", 5) &&
+      strcmp(name, ACL_EA_ACCESS) &&
+      strcmp(name, ACL_EA_DEFAULT))
```

Comment:
Why not return `CEPHFS_ENODATA` around here ?

Reply:
Sorry, I didn't get that question.

Our intention here is to catch only `CEPHFS_ENODATA` after _removexattr()_ is completed and return success to match kernel behaviour. An early return of any sorts would skip other possible errors during _removexattr()_. Does that make sense?

Reply:
Sorry, I missed reading this and please just ignore this.